parquet-tools/src/main/java/org/apache/parquet/tools/command/MergeCommand.java

```diff
@@ -63,28 +104,70 @@ public MergeCommand() {
   @Override
   public String getCommandDescription() {
     return "Merges multiple Parquet files into one. " +
-      "The command doesn't merge row groups, just places one after the other. " +
+      "Without -b option the command doesn't merge row groups, just places one after the other. " +
       "When used to merge many small files, the resulting file will still contain small row groups, " +
-      "which usually leads to bad query performance.";
+      "which usually leads to bad query performance. " +
+      "To have adjacent blocks(row groups) merged together use -b option. " +
+      "Blocks will be grouped into larger one until the upper bound is reached. " +
+      "Default block upper bound 128 MB and default compression SNAPPY can be customized using -l and -c options";
   }
 
   @Override
   public void execute(CommandLine options) throws Exception {
+    super.execute(options);
+
+    boolean mergeBlocks = options.hasOption('b');
+
     // Prepare arguments
     List<String> args = options.getArgList();
-    List<Path> inputFiles = getInputFiles(args.subList(0, args.size() - 1));
+    List<Path> files = getInputFiles(args.subList(0, args.size() - 1));
     Path outputFile = new Path(args.get(args.size() - 1));
-
     // Merge schema and extraMeta
-    FileMetaData mergedMeta = mergedMetadata(inputFiles);
-    PrintWriter out = new PrintWriter(Main.out, true);
-
-    // Merge data
+    ParquetMetadata parquetMetadata = mergedMetadata(files);
     ParquetFileWriter writer = new ParquetFileWriter(conf,
-            mergedMeta.getSchema(), outputFile, ParquetFileWriter.Mode.CREATE);
+      parquetMetadata.getFileMetaData().getSchema(), outputFile, ParquetFileWriter.Mode.CREATE);
+    PrintWriter stdOut = new PrintWriter(Main.out, true);
```

Comment:
Unlike cat/dump command, these kinds of parquet tools could be used by some automation jobs. In that case, using log4j might be better than PrintWriter.